src/service/extensions-impl.js

```diff
@@ -248,12 +248,10 @@ export class Extensions {
       /* includeInserted */ false
     );
     devAssert(el, 'Cannot find script for extension: %s', extensionId);
-    // "Disconnect" the old script element and extension record.
+    // The previously awaited extension loader must not have finished or
+    // failed.
     const holder = this.extensions_[extensionId];
-    if (holder) {
-      devAssert(!holder.loaded && !holder.error);
-      delete this.extensions_[extensionId];
```

Comment:
We were deleting this so that `isExtensionScriptRequired_` would return true. If we don't delete the holder, we must set `scriptPresent` to `false`

Reply:
I see. I'll review the tests again. Someone should have failed.

Reply:
Fixed